core/src/main/java/ai/timefold/solver/core/impl/solver/AbstractSolver.java

```diff
@@ -40,26 +41,30 @@ public abstract class AbstractSolver<Solution_> implements Solver<Solution_> {
     protected final BestSolutionRecaller<Solution_> bestSolutionRecaller;
     // Note that the DefaultSolver.basicPlumbingTermination is a component of this termination.
     // Called "solverTermination" to clearly distinguish from "phaseTermination" inside AbstractPhase.
-    protected final Termination<Solution_> solverTermination;
+    protected final UniversalTermination<Solution_> solverTermination;
```

Comment:
Maybe rename field to `universalTermination`? Since there is a class called `SolverTermination` now.

Reply:
The comment explains the field name well, I think.
I agree that the seeming conflict with `SolverTermination` is not ideal, but the field name is correct - the termination will terminate the solver.